File: java/src/main/java/io/fivexer/sdk/model/WorkerMetricsToday.java
package io.fivexer.sdk.model;

/**
 * A worker's own metrics for today.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerMetricsToday {
    private String workerId;
    private String since;
    private int completedTasks;
    private int breakCount;
    private long totalBreakMs;
    private long longestBreakMs;
    private long workingMs;
    private long onShiftMs;
    private int shiftCount;

    public String getWorkerId() { return workerId; }
    public String getSince() { return since; }
    public int getCompletedTasks() { return completedTasks; }
    public int getBreakCount() { return breakCount; }
    public long getTotalBreakMs() { return totalBreakMs; }
    public long getLongestBreakMs() { return longestBreakMs; }

    /**
     * Real worked time (shift time minus breaks) once the shift log has today's rows; a worker
     * predating it keeps the old since-midnight approximation.
     */
    public long getWorkingMs() { return workingMs; }

    /** Time on shift today, breaks included. Additive — zero on servers predating the shift log. */
    public long getOnShiftMs() { return onShiftMs; }

    /** Additive — zero on servers predating the shift log. */
    public int getShiftCount() { return shiftCount; }
}

File: php/src/Model/WorkerMetricsToday.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A worker own metrics for today.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerMetricsToday
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $since,
        public readonly int $completedTasks,
        public readonly int $breakCount,
        public readonly int $totalBreakMs,
        public readonly int $longestBreakMs,
        /**
         * Real worked time (shift time minus breaks) once the shift log has today's rows; a
         * worker predating it keeps the old since-midnight approximation.
         */
        public readonly int $workingMs,
        /** Time on shift today, breaks included. Additive — 0 on servers predating the shift log. */
        public readonly int $onShiftMs = 0,
        /** Additive — 0 on servers predating the shift log. */
        public readonly int $shiftCount = 0,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            since: (string) ($data['since'] ?? ''),
            completedTasks: (int) ($data['completedTasks'] ?? 0),
            breakCount: (int) ($data['breakCount'] ?? 0),
            totalBreakMs: (int) ($data['totalBreakMs'] ?? 0),
            longestBreakMs: (int) ($data['longestBreakMs'] ?? 0),
            workingMs: (int) ($data['workingMs'] ?? 0),
            onShiftMs: (int) ($data['onShiftMs'] ?? 0),
            shiftCount: (int) ($data['shiftCount'] ?? 0),
        );
    }
}
